[lockfree-queue]: GHC 9.4 support

I am seeing this error while building lockfree-queue with GHC 9.4.2:

```
Building library for lockfree-queue-0.2.3.1..
[1 of 2] Compiling Data.Concurrent.Queue.MichaelScott ( Data/Concurrent/Queue/MichaelScott.hs, dist/build/Data/Concurrent/Queue/MichaelScott.o, dist/build/Data/Concurrent/Queue/MichaelScott.dyn_o )

Data/Concurrent/Queue/MichaelScott.hs:42:24: error:
    Not in scope: ‘GPW.sameMutVar#’
    NB: the module ‘GHC.PrimopWrappers’ does not export ‘sameMutVar#’.
    Suggested fix:
      Perhaps use one of these:
        ‘GPW.casMutVar#’ (imported from GHC.PrimopWrappers),
        ‘GPW.takeMVar#’ (imported from GHC.PrimopWrappers),
        ‘GPW.newMutVar#’ (imported from GHC.PrimopWrappers)
   |
42 | sameMutVar# x y = case GPW.sameMutVar# x y of { 0# -> False; _ -> True }
   |                        ^^^^^^^^^^^^^^^
```

I am wondering if this patch fixes this issue: https://gitlab.haskell.org/ghc/head.hackage/-/blob/master/patches/lockfree-queue-0.2.3.1.patch .

--- a/lockfree-queue/Data/Concurrent/Queue/MichaelScott.hs
+++ b/lockfree-queue/Data/Concurrent/Queue/MichaelScott.hs
@@ -35,13 +35,13 @@ import Data.Atomics (readForCAS, casIORef, Ticket, peekTicket)
 
 -- GHC 7.8 changed some primops
 import GHC.Base  hiding ((==#), sameMutVar#)
-import GHC.Prim hiding ((==#), sameMutVar#)
-import qualified GHC.PrimopWrappers as GPW
+import GHC.Exts hiding ((==#), sameMutVar#)
+import qualified GHC.Exts as Exts
 (==#) :: Int# -> Int# -> Bool
-(==#) x y = case x GPW.==# y of { 0# -> False; _ -> True }
+(==#) x y = case x Exts.==# y of { 0# -> False; _ -> True }
 
 sameMutVar# :: MutVar# s a -> MutVar# s a -> Bool
-sameMutVar# x y = case GPW.sameMutVar# x y of { 0# -> False; _ -> True }
+sameMutVar# x y = case Exts.sameMutVar# x y of { 0# -> False; _ -> True }
 
 
 -- Considering using the Queue class definition:

